packages/vscode-extension/src/webview/views/LaunchConfigurationView.tsx

```diff
@@ -1,57 +1,152 @@
+function SingleSelect({
+  initialValue,
+  ...props
+}: React.ComponentProps<typeof VscodeSingleSelect> & { initialValue?: string }) {
+  const [value, setValue] = useState(initialValue);
+  return (
+    <VscodeSingleSelect
+      value={value}
+      onChange={(e) => setValue((e.target as HTMLSelectElement).value)}
+      {...props}
+    />
+  );
 }
 
-function LaunchConfigurationView({ launchConfigToUpdate }: LaunchConfigurationViewProps) {
-  const { openModal, closeModal } = useModal();
-  const applicationRoots = useApplicationRoots();
+function getLaunchConfigAttrs() {
+  const radonIDEDebugger = extensionPackageJSON.contributes?.debuggers?.find(
+    (config) => config.type === "radon-ide"
+  );
+  return radonIDEDebugger?.configurationAttributes?.launch;
+}
 
-  const { project, projectState } = useProject();
+function undefinedIfEmpty(value: string) {
+  return value === "" ? undefined : value;
+}
 
-  const isEditingSelectedConfig = useMemo(
-    () => _.isEqual(projectState.selectedLaunchConfiguration, launchConfigToUpdate),
-    [projectState.selectedLaunchConfiguration, launchConfigToUpdate]
-  );
+function undefinedIfAuto(value: string) {
+  return value === "auto" ? undefined : value;
+}
 
-  const [newLaunchConfigOptions, setNewLaunchConfigOptions] = useState<LaunchConfigurationOptions>(
-    launchConfigToUpdate ? optionsForLaunchConfiguration(launchConfigToUpdate) : {}
-  );
-  const { android, appRoot, ios, eas, isExpo, metroConfigPath } = newLaunchConfigOptions;
-  const { xcodeSchemes, easBuildProfiles } = useAppRootConfig(appRoot ?? applicationRoots[0]?.path);
+function serializeLaunchConfig(formData: FormData) {
+  const data = Object.fromEntries(formData as any);
+  const newConfig: LaunchConfigurationOptions = {
+    name: data.name ?? undefined,
+    appRoot: undefinedIfAuto(data.appRoot),
+    metroConfigPath: undefinedIfEmpty(data.metroConfigPath),
+    isExpo: data.isExpo === "true" ? true : data.isExpo === "false" ? false : undefined,
+  };
 
-  function update<K extends keyof LaunchConfigurationOptions>(
-    key: K,
-    value: LaunchConfigurationOptions[K] | "Auto"
-  ) {
-    if (value === "Auto") {
-      value = undefined;
+  for (const platform of ["ios", "android"] as const) {
+    const buildType = data[`buildType.${platform}`];
+    if (buildType === "standard") {
+      if (platform === "ios") {
+        newConfig.ios = {
+          scheme: undefinedIfAuto(data["ios.scheme"]),
+          configuration: undefinedIfEmpty(data["ios.configuration"]),
+        };
+      } else if (platform === "android") {
+        newConfig.android = {
+          buildType: undefinedIfEmpty(data["android.buildType"]),
+          productFlavor: undefinedIfEmpty(data["android.productFlavor"]),
+        };
+      }
+    } else if (buildType === "custom") {
+      newConfig.customBuild = {
+        ...newConfig.customBuild,
+        [platform]: {
+          buildCommand: data[`customBuild.${platform}.buildCommand`],
```

Comment:
We may want to check if this (and other required values) are empty, and disable the `save` button in that case.